bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/loader/AbstractScriptFileWatcher.java

```diff
@@ -378,16 +375,25 @@ public void onDependencyChange(String scriptIdentifier) {
 
     @Override
     public synchronized void onReadyMarkerAdded(ReadyMarker readyMarker) {
+        int newLevel = Integer.parseInt(readyMarker.getIdentifier());
+
+        if (newLevel <= currentStartLevel) {
+            logger.trace("onReadyMarkerAdded({}) - ignored {}", readyMarker, watchPath);
+            return;
+        }
+
         int previousLevel = currentStartLevel;
-        currentStartLevel = Integer.parseInt(readyMarker.getIdentifier());
+        currentStartLevel = newLevel;
+
+        logger.trace("onReadyMarkerAdded({}) {} => {} {}", readyMarker, previousLevel, currentStartLevel, watchPath);
```

Comment:
I've changed it as suggested. However, I added the watchPath because when you have multiple instances of this class, you can't tell which is which otherwise.